bin/sdk-cli.js

```diff
@@ -43,7 +43,14 @@ const getBaseConfig = ( options = {} ) => {
 };
 
 const build = ( target, argv ) => {
-	const config = target.config( { argv, getBaseConfig, calypsoRoot } );
+		const targetConfig =target.config( { argv, getBaseConfig, calypsoRoot } );
+	const config = {
+		...targetConfig,
+		output: {
+			...targetConfig.output,
+			publicPath: '/wp-content/plugins/jetpack/_inc/blocks/',
```

Comment:
Can we generate this automatically out of the output dir?

Reply:
I don't think so. Output dir is filesystem, publicPath is where the resources can be requested from the web.

@dmsnell Brought up a great point:

> will it continue to work if a plugin causes URL rewrites for the WordPress install?
> the loading code generated with `plugin_url()` in PHP should have a guarantee, but we don’t have that from JS or from the build script

This isn't straightforward. If I'm building a bundle for a site under my control (like the CDN), I can set a fixed `publicPath`. But what about bundling assets for a plugin to be used on any WordPress site?

Reply:
https://webpack.js.org/guides/public-path/#on-the-fly might be of use .... somehow.

Reply:
Definitely! It appears we'd just need to enqueue the entry script after an inline script that correctly sets that global 💥

Reply:
Those docs are unclear to me. Do you think we could run an experiment where a separate JavaScript script sets that global variable at runtime?

That is, imagine loading a script as a WordPress dependency…

```php
wp_enqueue_script(
  'jetpack-code',
  plugins_url( '/dist/code.js' ),
  array( 'public-path-setter' )
);
```

further, with this hard global does that mean that only a single application can set the public path? if someone else were to try and do the same would we be fighting each other for that path?

I'm wondering if it would be possible to propose a new way of getting asset paths in `webpack` itself, like by providing a runtime function

```js
publicPath: {
  module: {
    path: 'client/app/boot',
    export: 'getAssetPath'
  }
}
```

This is tricky in several ways but if we could have `webpack` look to some runtime code to load our assets then we would have full freedom to support a variety of configurations and even to let multiple independent sub-apps do separate dynamic asset loading

Reply:
I'll experiment with this a bit and see how things work 👍